test/integration/pkg/generated/openapi_generated.go

```diff
@@ -25,10 +25,15 @@ package generated
 import (
 	spec "github.com/go-openapi/spec"
 	common "k8s.io/kube-openapi/pkg/common"
+	custom "k8s.io/kube-openapi/test/integration/testdata/custom"
 )
 
 func GetOpenAPIDefinitions(ref common.ReferenceCallback) map[string]common.OpenAPIDefinition {
 	return map[string]common.OpenAPIDefinition{
+		"k8s.io/kube-openapi/test/integration/testdata/custom.Bac":              common.EmbedOpenAPIDefinitionIntoV2Extension(custom.Bac{}.OpenAPIV3Definition(), custom.Bac{}.OpenAPIDefinition()),
```

Comment:
so what does it mean?

it means that in the future (once we have `oas v3`) we could have a way to keep and generate both `v2` and `v2 ` schemas (`spec.Schema`)  in just one place, is that correct?

do we expect them to be different (`v2` vs `v3`) ? 

from what I have seen they are quite compatible:
https://github.com/OAI/OpenAPI-Specification/blob/master/versions/3.0.0.md#schemaObject

https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md#schemaObject

Reply:
> it means that in the future (once we have oas v3) we could have a way to keep and generate both v2 and v2 schemas (spec.Schema) in just one place, is that correct?

exactly, but only for those types have differences in v2 and v3 schema. for now, it's basically meant for `IntOrString`. in v2 spec, we made walkaround for those types we have to denote in v3 grammar e.g.`anyOf`. the pull will keep those v2 walkarounds in the builder in the extension so that:
- __for v2 publishing endpoint__: we shadow the generated schema using the new extension.
- __for v3 publishing endpoint__: we filter out the extension from the generated schema.


```go

// OpenAPISchemaType is used by the kube-openapi generator when constructing
// the OpenAPI spec of this type.
//
// See: https://github.com/kubernetes/kube-openapi/tree/master/pkg/generators
func (IntOrString) OpenAPISchemaType() []string { return []string{"string"} }

// OpenAPISchemaFormat is used by the kube-openapi generator when constructing
// the OpenAPI spec of this type.
func (IntOrString) OpenAPISchemaFormat() string { return "int-or-string" }
```

for `IntOrString`, the v3 spec will be an `anyOf`